Add a text size preference/option to the Project Browser

runrevelanor opened the pull request:
A text size preference in the Project Browser allows the user to choose
the text size used for text in the Project Browser. This preference
only applies to the Project Browser.
- revidelibrary
  - add colors to be used for the connector and disclosure icons in the 
    Project Browser
- revPreferencesGUI 
  - add a combo box allowing the user to set the text size on the Project 
    Browser card
- revIDEProjectBrowser 
  - remove the row templates from the stack
- revProjectBrowserBehavior
  - add a text size preference to the header
  - register for changes to the Project Browser text size pref
  - handle the Project Browser text size pref
  - set the line height in the Data View to take the text size
    pref into account
  - refer to the new row templates
  - add a function that will return the points of a line graphic
    used to display part of the tree structure
- add individual stacks holding the row templates in the templates
  folder. The templates now use graphics and svg widgets instead of
  buttons with icons to show the tree structure
- update the row behaviors
  - take the text size into account when drawing a row
  - update the graphics and svg widgets used to display the tree
    structure rather than setting the icon of buttons, allowing the
    row height to vary based on the text size

Closes #1096

runrevelanor committed:
Add a text size preference/option to the Project Browser

A text size preference in the Project Browser allows the user to choose
the text size used for text in the Project Browser. This preference
only applies to the Project Browser.

- revidelibrary - add colors to be used for the connector and
disclosure icons in the Project Browser
- revPreferencesGUI - add a combo box allowing the user to set the
text size on the Project Browser card
- revIDEProjectBrowser - remove the row templates from the stack
- revProjectBrowserBehavior
	- add a text size preference to the header
	- register for changes to the Project Browser text size pref
	- handle the Project Browser text size pref
	- set the line height in the Data View to take the text size
	pref into account
	- refer to the new row templates
	- add a function that will return the points of a line graphic
	used to display part of the tree structure
- add individual stacks holding the row templates in the templates
folder. The templates now use graphics and svg widgets instead of
buttons with icons to show the tree structure
- update the row behaviors
	- take the text size into account when drawing a row
	- update the graphics and svg widgets used to display the tree
	structure rather than setting the icon of buttons, allowing the
	row height to vary based on the text size

livecodeali commented:
@elanor Can you set the margins of the object name fields to 8? At the moment they are something like 6,10,6,3 which causes issues with the placement of the text for me at least.

Also margins of 6 for the number of lines of code labels seems to work better for me, but maybe that won't be the same on Windows? I'm not sure.

runrevelanor committed:
Improve the appearance of the Project Browser

The margins of the field in each row template for the Project Browser
are set to standard margins and each row behavior has been updated to
lay out the controls taking the text size preference into account.

The revPreferencesGUI stack has been updated as it was resetting the
Project Browser Text Size preference to default when opened.

livecodeali commented:
@livecode-vulcan review ok e5cfd4c

livecode-vulcan commented:
💙 review by @livecodeali ok e5cfd4c9e77de93fac389425e8cb0cbf5cf4b4e4
